movies/models.py

```diff
@@ -1,26 +1,68 @@
 from django.db import models
 from django.contrib.auth.models import User
 from django.utils.functional import cached_property
-from cinema import settings
-from cinemas.models import Hall
-from datetime import timedelta
+from cinema import settings as base_settings
+from django.core.files.uploadedfile import SimpleUploadedFile
+from PIL import Image
+import io
+import os
 
 # Create your models here.
 
 
 class Movie(models.Model):
     user = models.ForeignKey(User, null=True, on_delete=models.CASCADE)
     name = models.CharField(max_length=100, null=True, verbose_name='Title')
-    poster = models.URLField(verbose_name='Poster')
+    poster = models.ImageField(verbose_name='Poster', max_length=500)
     description = models.TextField(max_length=200, null=True, verbose_name='Description', blank=True)
     imdb_id = models.CharField(max_length=30, null=True, verbose_name='IMDB_id')
     length_min = models.IntegerField(verbose_name='Length')
     trailer_url = models.URLField(max_length=200, null=True, blank=True)
     is_scheduled = models.BooleanField(default=False)
+    thumbnail = models.ImageField(upload_to='thumbnails', max_length=500, null=True, blank=True)
 
     def __str__(self):
         return self.name
 
     @cached_property
     def get_imdb_url(self):
-        return f'{settings.IMDB_URL}{self.imdb_id}'
+        return f'{base_settings.IMDB_URL}{self.imdb_id}'
+
+    def create_thumbnail(self, size=(128, 128)):
+
+        if not self.poster:
+            return
+
+        poster_type = self.poster.file.content_type
+
+        if poster_type == 'image/jpeg':
```

Comment:
A best practice is to make a general rule for image processing. There are numerous image extensions out there (jpeg / jpg, png, gif, tiff, psd, raw, bmp, webp just to name a few). Treating the cases explicitly may cause you to forget about some types.